src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,381 @@
+        for (final Colony colony : colonyList)
+        {
+            colonyIdList.add(String.valueOf(colony.getID()));
+        }
+        return colonyIdList;
+    }
+
+    private List<String> getCitizenNames()
+    {
+        final List<Colony> colonyList = ColonyManager.getColonies();
+        final List<String> citizenNameList = new ArrayList<>();
+        for (final Colony colony : colonyList)
+        {
+            final List<CitizenData> citizenDataList = colony.getCitizenManager().getCitizens();
+            for (final CitizenData citizenData : citizenDataList)
+            {
+                citizenNameList.add(citizenData.getName());
+            }
+        }
+        return citizenNameList;
+    }
+
+    public List<String> getTabCompletions(@NotNull final MinecraftServer server,
+            @Nullable final BlockPos pos,
+            final String potentialArgumentValue)
+    {
+        switch (this)
+        {
+            case INTEGER:
+                return Collections.emptyList();
+            case BOOLEAN:
+                return Arrays.asList(new String[] {"true", "false"});
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                return getCoordinateTabCompletions(pos, potentialArgumentValue);
+            case PLAYER:
+                final List<String> playerNameStrings = getOnlinePlayerNames(server);
+                return playerNameStrings.stream().filter(k -> k.startsWith(potentialArgumentValue)).collect(Collectors.toList());
+            case COLONY:
+                return getColonyTabCompletions(potentialArgumentValue);
+            case CITIZEN:
+                return getCitizenTabCompletions(potentialArgumentValue);
+            default:
+                throw new IllegalStateException("Unimplemented ActionArgumentType tab completion");
+        }
+    }
+
+    private List<String> getCoordinateTabCompletions(@Nullable final BlockPos pos, final String potentialArgumentValue)
+    {
+        if (null == pos)
+        {
+            return Collections.emptyList();
+        }
+        if (potentialArgumentValue.isEmpty())
+        {
+            switch (this)
+            {
+                case COORDINATE_X:
+                    return Collections.singletonList(String.valueOf(pos.getX()));
+                case COORDINATE_Y:
+                    return Collections.singletonList(String.valueOf(pos.getY()));
+                case COORDINATE_Z:
+                    return Collections.singletonList(String.valueOf(pos.getZ()));
+                default:
+                    // We will never reach here.
+                    break;
+            }
+            return Collections.emptyList();
+        }
+        return Collections.emptyList();
+    }
+
+    private List<String> getColonyTabCompletions(final String potentialArgumentValue)
+    {
+        // TODO: use the colony we are in as the default tab completion.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)